src/core/mainloop/mainloop.c

```diff
@@ -1575,14 +1558,15 @@ initialize_periodic_events(void)
 
   periodic_events_initialized = 1;
 
-  /* Set up all periodic events. We'll launch them by roles. */
-  int i;
-  for (i = 0; periodic_events[i].name; ++i) {
-    periodic_event_setup(&periodic_events[i]);
+  for (int i = 0; mainloop_periodic_events[i].name; ++i) {
+    periodic_events_add(&mainloop_periodic_events[i]);
   }
 
+  /* Set up all periodic events. We'll launch them by roles. */
+  periodic_events_setup_all();
```

Comment:
Should this be done by the subsystem that is setup its own event?

Reply:
See note above about documentation commit; I think that will address this issue too